kobo/apps/superuser_stats/tasks.py

```diff
@@ -174,6 +411,79 @@ def get_row_for_user(u: KobocatUser) -> list:
 
 
 @shared_task
+def generate_user_statistics_report(
+        output_filename: str,
+        start_date: str,
+        end_date: str
+):
+    data = []
+
+    asset_queryset = Asset.objects.values('owner_id', 'owner__extra_details').filter(
+        asset_type=ASSET_TYPE_SURVEY,
+        date_created__range=(start_date, end_date)
+    )
+    records = asset_queryset.annotate(
+        form_count=Count('pk')
+    ).order_by()
+    forms_count = {
+        record['owner_id']: record['form_count'] for record in records
+    }
+
+    # Filter the asset_queryset for active deployments
+    asset_queryset = asset_queryset.filter(_deployment_data__active=True)
+    records = asset_queryset.annotate(
+        deployment_count=Count('pk')
+    ).order_by()
+    deployment_count = {
+        record['owner_id']: record['deployment_count']
+        for record in records
+    }
+
+    # Get records from SubmissionCounter
+    records = (
+        ReadOnlyKobocatMonthlyXFormSubmissionCounter.objects.annotate(
+            date=Cast(
+                Concat(F('year'), Value('-'), F('month'), Value('-'), 1),
+                DateField(),
+            )
+        ).filter().values(
+            'user_id',
+            'user__username',
+            'user__date_joined',
+        ).order_by('user__date_joined').annotate(count_sum=Sum('counter'))
+    )
+
+    for record in records:
+        user_details = ExtraUserDetail.objects.get(
+            user__username=record['user__username']
```

Comment:
little optimization, No need to create a `JOIN` since we already have the `user_id`.

```python 
 user_details = ExtraUserDetail.objects.get(
            user_id=record['user_id']
            ...
```